src/Microsoft.AspNetCore.Authentication/ClaimResolver.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Security.Claims;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public abstract class ClaimResolver<T>
```

Comment:
Is this going to be used outside of OAuth?  Should we target this against OAuth/JObject for now?